starknet_py/net/client_errors.py

```diff
@@ -19,17 +22,16 @@ class ContractNotFoundError(ClientError):
 
     def __init__(
         self,
-        block_hash: Optional[Union[int, str]] = None,
-        block_number: Optional[int] = None,
+        address: AddressRepresentation,
+        block_hash: Optional[Hash] = None,
+        block_number: Optional[Union[int, Tag]] = None,
     ):
-        require_block_identifier(block_hash, block_number)
-
+        is_identifier = block_hash is not None or block_number is not None
         identifier = block_hash or block_number
-        self.identifier = str(identifier) if isinstance(identifier, int) else identifier
-
-        super().__init__(message=f"No contract found for identifier: {self.identifier}")
+        identifier_name = "block_hash" if block_hash else "block_number"
```

Comment:
I wonder if we should also handle "latest" and "pending, since we're doing a bit of extra logic with these: In the GatewayClient if someone uses parameter `block_hash="latest"`, we're actually making a request with `{"block_number": "latest}`.

Reply:
Well, it's a bug on our side. `block_hash` hash to be an actual hash and not any literal, as we allow in many places in code.
But I guess it'd be a small breaking change and it's not that harmful to leave it.